### DISCUSSION TITLE
Can I create blog entries as directories?

### DISCUSSION THREAD

**Discussion by hhimanshu** [facebook/docusaurus#5301] [Q&A] [answered]
Currently, as per the [documentation](https://docusaurus.io/docs/blog), the blog entries structure looks like

```
blog/
       2021-08-01.md
       2021-08-03.md
       2021-08-04.md
```

Is it possible to have the blogs as directory structures like below?

```
blog/
       2021-08-01/
                           index.md
                           _partial.md
                          images/
                                      one.jpeg
                                      two.jpeg

      2021-08-04/
                           index.md
                           _partial-1.md
                          images/
                                      three.jpeg
                                      four.jpeg
```

Benefits?
All blog related assets - image, css, extra components, mdx files can be localized with the blog. This can sprinkle the assets, code elsewhere in the codebase

**Comment by slorber** [comment, marked as answer]
I think it might work, but I'm not sure we'd read the date from the folder name in such case, and you may need to provide the date as frontmatter.

Just try and we'll see, otherwise this make sense that we support that use-case

**Reply by hhimanshu** [reply]
Man! you are fast ⏩ 
I will give it a try and get back

**Reply by hhimanshu** [reply]
@slorber, I gave it a try, but it doesn't show up in `/blog` list. The blog is at https://github.com/hhimanshu/docusaurus-101/blob/blog-directory/blog/5-Aug-2021/index.md (separate branch)

Am I missing anything?

**Reply by slorber** [reply]
you are right it's not supported, just a bad glob import using `*.{md,mdx}` instead of `**/*.{md,mdx}`. You can make it work with the "include" option but I'm adding support by default right now

**Reply by slorber** [reply]
https://github.com/facebook/docusaurus/pull/5309

**Reply by hhimanshu** [reply]
Fantastic, thank you @slorber for considering this feature. It will be indeed useful. Let me know once it is merged, and I will give it a try on my side

**Reply by slorber** [reply]
👍 

to know when it's merged, just subscribe to the PR above, and then test with canary: https://docusaurus.io/community/canary

**Reply by slorber** [reply]
just merged

**Reply by Ouch1978** [reply]
I'm looking for this feature, too.
In my case, I would like to have both markdown file and all related static files stored in the same folder.
For ex:
blog/
       2021-08-01-my-blog-post-01/
                           2021-08-01-my-blog-post-01.md
                           one.jpeg
                           two.jpeg
Is it possible, too?

**Reply by Ouch1978** [reply]
I did a test with the canary build.
Seems it's working well and I could achieve what I want.

**Reply by slorber** [reply]
Thanks for the feedback :)
